Android app

karteekdhara98 opened the pull request:
Please check if POST has been done in the right way.

karteekdhara98 committed:
Add files via upload

kunalgrover05 commented on ServerData.java:
```diff
@@ -0,0 +1,1114 @@
+
+    /**
+     *
+     * @param point50
```
This is no way to declare 50 points. Please do it in a proper way.
This is how the app will work
1- App makes request to server to get all nodes. Server returns a list of points as response.
2- App sends request with initial point and final point, Server returns shortest path from initial to final point.

karteekdhara98 replied:
In ServerData.java I'll replace those fifty points with a list.
For the rest part:
From what I've understood this is what the current code is doing. Please correct me if I am wrong. Using the @POST i am posting the initial and final points as entered by the user and the @GET gets the nodes which will make up the shortest path. But I am doing both using fields in the same ServerData class.
So the second part is done right?
As for the  first part I will have to  implement a second @GET in onCreate(). I will update the code asap.

kunalgrover05 replied:
Your understanding about GET/POST is incorrect. Please read up on how Requests/Responses work. You will have total of 2 requests:
1- Get list of 50 points- Request parameters: None, Response: List of points
2- Get shortest path from A to B- Request parameters: A,B, Response: Shortest path
Both of them can be GET requests and you don't need a POST request as such.

karteekdhara98 replied:
So this is what i have changed till now.
1) Removed those fifty points and made a list out of it(So we don't have an upper limit on nodes). Also i removed the initial and final point fields from ServerData.java
2) In MapsActivity.java i replaced all those lines of putting markers with for loops.
3) I have removed the post request. Added a GET for the node list with the method being 
called in onCreate(). 
4) Additionally I have added query parameters in the existing GET. The  parameters are two Double lists. One for initial and one for final point.
5) And for both these asynchronous requests the callback class is  ServerData.java
Please tell me if anything else needs to be changed.

kunalgrover05 replied:
Sounds good. Have you tested your app?

karteekdhara98 replied:
I checked if the markers were being placed on the map,if a blue dot was
shown at the current location and if the initial and final markers were
being updated. However I couldn't think of a way to check if the rest part
was working properly. Also what about the destination url for both these
requests? For now I just put one url.
I will push this code in a while. Tomorrow I will work on camera in the map
and I was thinking that since this is mostly done I will pull out a map a
iitm and try to identify nodes so we can decide on the number of devices we
will need.
On Saturday, May 21, 2016, Kunal Grover notifications@github.com wrote:

> In ServerData.java:
> 
> > - public void setPoint49(List<Double> point49) {
> > - this.point49 = point49;
> > - }
> >   +
> > - /**
> > - *
> > - \* @return
> > - \* The point50
> > - */
> > - public List<Double> getPoint50() {
> > - return point50;
> > - }
> >   +
> > - /**
> > - *
> > - \* @param point50
> 
> Sounds good. Have you tested your app?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub<
> https://ci6.googleusercontent.com/proxy/8w_4anAKOgcXa9cF0bvLLDi1ueC8syBYjJag8alFEFGMs_SNMqB0vzzDN5rsAHT2C4VwLcd_tyHjEIRf0CuHTVZFGxGWD-yByjWgDemW8UGhg3uFsDduGnfJBl6YSkCjUZbS_JmbCBE7uuA_-k5EQ3oCj2q6Jg=s0-d-e1-ft#https://github.com/notifications/beacon/ARhz6DeQH7did0s2p8pfds9hxgI1Tkhoks5qDxjkgaJpZM4IjuFm.gif

## 

Karteek Dhara
First year student,
Dual Degree | Dept. of Electrical Engineering
Indian Institute of Technology, Madras

karteekdhara98 committed:
Update HttpRequest.java

Changes made:
1) POST was removed and was replaced with query params, i.e. the initial and final points are now passed as query parameters.
2)Another GET was added to take in list of all node locations

karteekdhara98 committed:
Update MapsActivity.java

Check onCreate()

karteekdhara98 committed:
Update ServerData.java

1)Removed initial and final point fields as now they are passed as query parameters in the GET request.
2)Replaced the fifty points with a list. So now  there is no upper limit on nodes.
